core/dbt/tests/util.py

```diff
@@ -86,22 +89,27 @@ def run_dbt(args: List[str] = None, expect_pass=True):
 # If you want the logs that are normally written to a file, you must
 # start with the "--debug" flag. The structured schema log CI test
 # will turn the logs into json, so you have to be prepared for that.
-def run_dbt_and_capture(args: List[str] = None, expect_pass=True):
+def run_dbt_and_capture(args: List[str] = None, expect_pass=True, capture_stderr=False):
     try:
         stringbuf = StringIO()
         capture_stdout_logs(stringbuf)
-        res = run_dbt(args, expect_pass=expect_pass)
-        stdout = stringbuf.getvalue()
+        if capture_stderr:
+            with redirect_stdout(stringbuf):
```

Comment:
Lots of `dbt debug` wasn't being piped out to `stdout` for whatever reason. I think we need this trick in our toolkit for future use cases.

Reply:
Potentially relevant: https://github.com/dbt-labs/dbt-core/issues/6350

Reply:
I think I'm the one that added [`redirect_stdout`](https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stdout) here originally.

If I remember correctly, it was _only_ added in order to capture the print statements from `dbt debug` for the TODO test (see my other PR comment below on that test).

If all the print statements are removed, we might not need the `capture_stderr` parameter and the associated logic. We'd have to check one way or the other.

If we can remove it altogether to simplify, then great!

If we keep it, I'd suggest renaming `capture_stderr` to `redirect_stdout` (or some other alternative) -- the former feels like a misnomer.

Reply:
I would rather that we didn't add an additional argument to run_dbt_and_capture if we can avoid it.

Reply:
As I recall, I had to override this to capture the output because before other fixes in this PR, the `debug` output was being kicked out to err as warnings. I'll double check if that's still relevant now because this change may no longer be needed (which would be nice)

Reply:
Checked it! This is _no longer needed_ 🎉 It was needed when I started (before I changed the text of `dbt list` to no longer be a warning, which gets piped to `stderr`). Great catch